### PR TITLE
Add some Rusty types to wrap around various maps

### PR DESCRIPTION
**Pull request by laarmen**
I started using separate types for various types of maps. This is by no means exhaustive, but it's a start.

This will probably conflict with #1 , I'll rebase if necessary.

### CODE REVIEW

**Comment by uccidibuti** [comment]
Thank you for these improvements!

File: rebpf/src/xdp.rs [hunk 1]
```diff
@@ -29,6 +29,9 @@ pub enum XdpAction {
     REDIRECT = libbpf::xdp_action_XDP_REDIRECT,
 }
 
+#[repr(transparent)]
+pub struct XdpMetadata(libbpf::xdp_md);
```
Comment (uccidibuti) [inline rebpf/src/xdp.rs:33]: Is it a problem for you if I change the name of the struct in XdpMd? I prefer keep the same name of libbpf library for struct/functions that wrapping libbpf..

**Comment by laarmen** [comment]
Well, I personally prefer explicit names for my Rust code, contrary to C
coders who prefer brevity, but it's your project, you're the boss 😉

Le sam. 18 avr. 2020 à 17:40, Lorenzo Vannucci <notifications@github.com> a
écrit :

**Comment by uccidibuti** [comment]
I agree with you but in rebpf for only the functions/structs that wrap libbpf I want to mantain the same names.. I did also a small restyle, in details:
- I have moved all xdp.rs functions/structs into lib.rs, I think that is better put all wrapped libbpf codes into lib.rs, helpers.rs and high level abstraction api in other module (i.e. maps.rs). 
- I have improved error management.
- I have switched the license from LGPL to MIT.
- I have opened tasks to do in roadmap as issues with "roadmap" label.
Thanks you again for your improvements, I hope to recive other pull request and that this small restyle will not a problem for you..

**Pull request merged**